### PR TITLE
gh-63301: Set exit code when tabnanny CLI exits on error

### PR DESCRIPTION
**Pull request by ultimatecoder**
Setting appropriate `exit` code when `tabnanny` is exiting with errors. Occurrences of `errprint` under `chec()` are escaped with a reason to continue traversing files under a directory.

* Issue: gh-63301

### CODE REVIEW

**Comment by vstinner** [comment]
You didn't patch  all calls to errprint(). Well, another approach is to put the exit() in errprint() directly.

Moreover, test_tabnanny fails.

**Review by hauntsaninja — APPROVED** [review]
Thank you, this looks straightforward enough

File: Lib/tabnanny.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ def errprint(*args):
         sys.stderr.write(sep + str(arg))
         sep = " "
     sys.stderr.write("\n")
+    sys.exit(1)
```
Comment (arhadthedev) [inline Lib/tabnanny.py:38]: Can the whole `errprint()` be replaced with this:

```python
def errprint(*args):
    sys.exit(' '.join(str(arg) for arg in args))
```

?

**Comment by hauntsaninja** [comment]
Thank you Jaysinh and Erlend!

**Pull request merged**